enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -348,6 +349,29 @@ func (h *executorHandle) Serve(ctx context.Context) error {
 						log.CtxWarningf(ctx, "Could not re-enqueue task reservation for executor %q going down: %s", executorID, err)
 					}
 				}
+			} else if req.GetAskForMoreWorkRequest() != nil {
+				log.CtxDebugf(ctx, "Executor %q requested more work.", executorID)
+				node := h.getRegistration()
+				poolKey := nodePoolKey{os: node.GetOs(), arch: node.GetArch(), pool: node.GetPool()}
+
+				timeSinceLastCall := time.Since(lastRequestForMoreWork)
```

Comment:
if the executor has been running tasks for several minutes then the initial backoff will be 1 minute I think? (so the 5s -> 1m backoff doesn't seem like it will happen very often, and in practice it will mostly just be a 1m poll?)

Reply:
The client resets its timer each time it receives new work

Reply:
I tried testing it locally: started the executor, let it sit for a bit requesting more work, executed one task, then let it sit for a bit again - it prints these logs:

```
2024/11/16 22:24:45.213 WRN Requesting more work
2024/11/16 22:24:55.216 WRN Requesting more work
2024/11/16 22:25:15.221 WRN Requesting more work
2024/11/16 22:25:55.233 WRN Requesting more work
2024/11/16 22:26:55.234 WRN Requesting more work
2024/11/16 22:27:13.803 WRN Executing task. execution_id=/uploads/6fbc9d3d-80eb-46c5-ad3d-72533cf8273e/blobs/9107569f06afc0fb529f6cec7bc6cf67625375c82bc619a2e8c80ca365131115/143 executor_host_id=c02fda4a-6bbe-402d-a27a-666fa0edfbad executor_id=a3ee676e-7ed6-4244-94fe-752f8b4c2b8d invocation_id=2a442b03-3e5b-40ac-acc5-c4a75f0e8451
2024/11/16 22:27:17.462 WRN Requesting more work
2024/11/16 22:28:01.918 WRN Requesting more work
2024/11/16 22:29:01.919 WRN Requesting more work
```

The timestamps at the beginning are expected - 10s gap, then 20s, then 40s, then 60s

The timestamps after the task gets executed are unexpected though, right? The initial delay is 44s (I'm not really sure why), then 60s. It should restart the backoff and be 10, 20, 40, 60 again, no?

Reply:
I fixed this by also resetting the timer server-side when work is finished.